zerver/openapi/markdown_extension.py

```diff
@@ -541,5 +548,42 @@ def render_title(self, function: str) -> List[str]:
         return title
 
 
+class ResponseDescriptionPreprocessor(Preprocessor):
+    def __init__(self, md: markdown.Markdown, config: Mapping[str, Any]) -> None:
+        super().__init__(md)
+        self.api_url = config["api_url"]
+
+    def run(self, lines: List[str]) -> List[str]:
+        done = False
+        while not done:
+            for line in lines:
+                loc = lines.index(line)
+                match = MACRO_REGEXP_RESPONSE_DESC.search(line)
+
+                if match:
+                    function = match.group(2)
+                    text = self.render_responses_description(function)
+                    # The line that contains the directive to include the macro
+                    # may be preceded or followed by text or tags, in that case
+                    # we need to make sure that any preceding or following text
+                    # stays the same.
+                    line_split = MACRO_REGEXP_RESPONSE_DESC.split(line, maxsplit=0)
+                    preceding = line_split[0]
+                    following = line_split[-1]
+                    text = [preceding, *text, following]
+                    lines = lines[:loc] + text + lines[loc + 1 :]
+                    break
+            else:
+                done = True
+        return lines
```

Comment:
I think it would be well worth deduplicating these extensions as a follow-up PR.  The `run` code is the same for all of these aside from the regex and the `render` function, so I think we could have a base class that we subclass with different render functions and `REGEX` properties and delete a lot of duplicate code.